main.go

```diff
@@ -256,7 +283,7 @@ func main() {
 			log.Info("Starting lease controller to report status")
 
 			leaseUpdater := lease.NewLeaseUpdater(
-				clientset,
+				kubernetes.NewForConfigOrDie(reconciler.TargetK8sConfig),
```

Comment:
Will this lease be on the managed cluster (like shown here), or will it be on the hosting cluster?

I think it would be hosting cluster, based on https://github.com/open-cluster-management-io/enhancements/tree/main/enhancements/sig-architecture/63-hosted-addon#guidance-for-transforming-default-mode-addon-to-hosted-mode
> 2. Refactor the code of the addon agent to accept a flag like managed-kubeconfig, and use the kubeconfig received with the flag to build the managed cluster kube client, and use this client to process resources on the managed cluster. This kubeconfig should be different from the local kubeconfig or in-cluster kubeconfig which is mainly used for leader election and addon lease.